plugins/thumbnails/server/worker.py

```diff
@@ -157,3 +159,20 @@ def attachThumbnail(file, thumbnail, attachToType, attachToId, width, height):
     }
 
     return ModelImporter.model('file').save(thumbnail)
+
+
+def getImage(extension, data):
+    """
+    Check extension of image and opens it.
+
+    :param extension: The extension of the image that needs to be opened.
+    :param data: The image file stream.
+    """
+    if 'dcm' in extension:
+        # Open the dicom image
```

Comment:
Need to check if this dicom file's pixel data is 2d or 3d and generate thumbnail accordingly.